fetch_request.go

```diff
@@ -308,22 +317,26 @@ func (o *fetchOpts) fetchMessages(channels map[string]interface{}) map[string][]
 					if d, ok := histResponse["message_type"]; ok {
 						switch v := d.(type) {
 						case float64:
-							histItem.MessageType = PNMessageType(int(v)).toMessageType()
+							histItem.MessageType = int(v)
 						case string:
 							t, err := strconv.ParseInt(v, 10, 64)
 							if err == nil {
-								histItem.MessageType = PNMessageType(int(t)).toMessageType()
+								histItem.MessageType = int(t)
 							} else {
-								o.pubnub.Config.Log.Printf("MessageType conversion error.")
+								o.pubnub.Config.Log.Printf("Type conversion error.")
 							}
 						default:
-							if o.WithMessageType {
-								histItem.MessageType = PNMessageTypeMessage.toMessageType()
+							o.pubnub.Config.Log.Printf("histResponse message_type type %vv", d)
+							if v != nil {
+								o.pubnub.Config.Log.Printf("histResponse message_type type %vv", reflect.TypeOf(v).Kind())
 							} else {
-								histItem.MessageType = ""
+								o.pubnub.Config.Log.Printf("histResponse message_type nil")
```

Comment:
Is that println expected?

Reply:
I'll check that

Reply:
Apparently this is how this code worked before. I'm not actually changing that. This is how it looks like straight on master